src/docs/markdown/caddyfile/directives.md

```diff
@@ -155,4 +159,33 @@ file_server
 acme_server
 ```
 
-You can override/customize this ordering by using the [`order` global option](/docs/caddyfile/options) or the [`route` directive](/docs/caddyfile/directives/route).
+
+
+## Sorting algorithm
+
+For ease of use, the Caddyfile adapter sorts directives according to the following rules:
+
+- Differently named directives are sorted by their position in the [default order](#directive-order#order). The default order can be overridden with the [`order` global option](/docs/caddyfile/options).
+
+- Same-named directives are sorted according to their [matchers](/docs/caddyfile/matchers#syntax).
+
+  - The highest priority is a directive with a single [path matcher](/docs/caddyfile/matchers#path-matchers).
+
+    Path matchers are sorted by specificity, from most specific to least specific.
+	
+	In general, this is performed by sorting by the length of the path matcher. There is one exception where if the path ends in a `*` and the paths of the two matchers are otherwise the same, the matcher with no `*` is considered more specific and sorted higher.
+
+    For example:
+    - `/foobar` is more specific than `/foo`
+    - `/foo` is more specific than `/foo*`
+    - `/foo/*` is more specific than `/foo*`
+
+  - A directive with any other matcher is sorted next, in the order it appears in the Caddyfile.
+
+    This includes path matchers with multiple values, and [named matchers](/docs/caddyfile/matchers#named-matchers).
+
+  - A directive with no matcher (i.e. matching all requests) is sorted last.
+
+- The [`vars`](/docs/caddyfile/directives/vars) directive has its ordering by matcher reversed, because it involves setting values which can overwrite eachother, so the most specific matcher should be evaluated last.
```

Comment:
I'd have to look but am sprinting through notifications atm -- is this also true of `map` and/or `header`? I can't recall off the top of my head so I just thought I'd make a quick note here.

Reply:
No we only put in that condition for `vars` currently. We could revisit for other directives though.

Reply:
Ok; let's leave as-is for now. I guess I even use the `header` example in my wiki article, Composing in the Caddyfile: https://caddy.community/t/composing-in-the-caddyfile/8291